### PR TITLE
feature: maintain battery in a certain range

### PR DESCRIPTION
**Pull request by tangb4c**
Allow battery tool to maintain battery between in a certain range.
Syntax: battery maintain min-max
Battery tool turn off charging above the maximum value, and on below the minimum value.
eg: battery maintain 20-70
eg: battery maintain 30-30

### CODE REVIEW

**Review by FlyfishO25 — CHANGES_REQUESTED** [review]
Well, in my understanding, this change actually make the computer only charge to the lower_threshold because when the battery level is lower than lower_threshold, it starts charging and stop it when it's higher than lower_thres, which make the actual behavior become "charge to $lower_threshold". I think the correct way should be "if lower than lower_threshold, charge to upper_threshold"

**Comment by Ben7on** [comment]
> stop it when it's higher than lower_thres

According to this if condition, It seems it will stop only once it's higher than upper_threshold

`if [[ "$battery_percentage" -ge "$upper_threshold" && "$is_charging" == "enabled" ]]; then`

**Comment by seefood** [comment]
Seems to me like a clone of #150, which includes some decent bash and applescript workarounds.

File: battery.sh [hunk 1]
```diff
@@ -487,8 +500,15 @@ if [[ "$action" == "maintain" ]]; then
 		exit 0
 	fi
 
+	if [[ "$setting" =~ ^[0-9]{1,3}-[0-9]{1,3}$ ]];then
```
Comment (seefood) [inline battery.sh:503]: Not sure how anyone would ever enter 100 or higher in either lower nor upper range, but here is an improvement suggestion:

^[0-9]{1,3}-(100|[0-9]{1,2})$

You could also do 1+[0-9]{1,2} and check against 101-199 but I thought it not as elegant :)

**Review by seefood — APPROVED** [review]

File: battery.sh (hunk 1 above)
Comment (tangb4c) [reply battery.sh:503]: You inspired me. 
All inputs that match the range mode should be matched, and user should be prompted when the number does not meet the requirements. That is better.

**Comment by tangb4c** [comment]
@actuallymentor do you have time to merge this MR?

**Comment by actuallymentor** [comment]
Thanks a lot for your work @tangb4c! And apologies for my late response. My response times vary a lot depending on how busy work is :)

The only request I have is to add a few extra comments explaining what your code does. I like to leave many comments so that new contributors can easily figure out what things do.

Please tag me when you added them and I will gratefully merge!

**Comment by FeeJai** [comment]
May I request a GUI feature for this too?

**Comment by seefood** [comment]
@tangb4c are you adding the remarks or would you like some 3rd party help with  an update PR? :)

**Comment by tangb4c** [comment]
> @tangb4c are you adding the remarks or would you like some 3rd party help with an update PR? :)

@seefood Do me a favor, pls, :D

**Comment by actuallymentor** [comment]
Closing so we can keep this central in #261, if that was in error let me know!

**Pull request closed without merging**